project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -662,26 +696,44 @@ record ProjectCreationResponse(String projectId) {
    *
    * @param projectId   the project's id
    * @param requestBody the information to be updated.
+   * @param requestId   the request ID, needs to be provided by the client and will be referenced in
+   *                    the response. If <code>null</code> or {@link String#isBlank()} is true, then
+   *                    a random UUID is assigned with {@link UUID#randomUUID()}.
    * @since 1.9.0
    */
   record ProjectUpdateRequest(String projectId, ProjectUpdateRequestBody requestBody,
-                              String id) implements CacheableRequest {
+                              String requestId) implements CacheableRequest {
+
+    public ProjectUpdateRequest {
+      if (projectId == null) {
+        throw new IllegalArgumentException("Project ID cannot be null");
+      }
+      if (projectId.isBlank()) {
+        throw new IllegalArgumentException("Project ID cannot be blank");
+      }
+      if (requestId != null && requestId.isBlank()) {
+        requestId = null;
+      }
+      if (requestId == null) {
+        requestId = UUID.randomUUID().toString();
+      }
```

Comment:
You can combine those 2 statements into
```suggestion
      if (requestId == null || requestId.isBlank()) {
        requestId = UUID.randomUUID().toString();
      }
```
This way you avoid temporal coupling for the `isBlank` case.